homeassistant/components/aftership/sensor.py

```diff
@@ -9,54 +9,109 @@
 from homeassistant.util import Throttle
 
-REQUIREMENTS = ['pyaftership==0.1.2']
+REQUIREMENTS = ["pyaftership==0.1.2"]
 
 _LOGGER = logging.getLogger(__name__)
 
-ATTRIBUTION = 'Information provided by AfterShip'
+ATTRIBUTION = "Information provided by AfterShip"
+ATTR_TRACKINGS = "trackings"
 
-CONF_SLUG = 'slug'
-CONF_TITLE = 'title'
-CONF_TRACKING_NUMBER = 'tracking_number'
+BASE = "https://track.aftership.com/"
 
-DEFAULT_NAME = 'aftership'
+CONF_SLUG = "slug"
+CONF_TITLE = "title"
+CONF_TRACKING_NUMBER = "tracking_number"
 
-ICON = 'mdi:package-variant-closed'
+DEFAULT_NAME = "aftership"
 
-MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
+ICON = "mdi:package-variant-closed"
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_API_KEY): cv.string,
-    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-})
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=5)
 
+SERVICE_ADD_TRACKING = "aftership_add_tracking"
+SERVICE_REMOVE_TRACKING = "aftership_remove_tracking"
 
-async def async_setup_platform(
-        hass, config, async_add_entities, discovery_info=None):
+ADD_TRACKING_SERVICE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_TRACKING_NUMBER): cv.string,
+        vol.Optional(CONF_TITLE): cv.string,
+        vol.Optional(CONF_SLUG): cv.string,
+    }
+)
+
+REMOVE_TRACKING_SERVICE_SCHEMA = vol.Schema(
+    {vol.Required(CONF_SLUG): cv.string,
+     vol.Required(CONF_TRACKING_NUMBER): cv.string}
+)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
     """Set up the AfterShip sensor platform."""
     from pyaftership.tracker import Tracking
 
     apikey = config[CONF_API_KEY]
     name = config[CONF_NAME]
-
     session = async_get_clientsession(hass)
     aftership = Tracking(hass.loop, session, apikey)
 
     await aftership.get_trackings()
 
-    if not aftership.meta or aftership.meta['code'] != 200:
-        _LOGGER.error("No tracking data found. Check API key is correct: %s",
-                      aftership.meta)
+    if not aftership.meta or aftership.meta["code"] != 200:
+        _LOGGER.error(
+            "No tracking data found. Check API key is correct: %s",
+            aftership.meta
+        )
         return
 
-    async_add_entities([AfterShipSensor(aftership, name)], True)
+    instance = AfterShipSensor(aftership, name)
+
+    async_add_entities([instance], True)
+
+    async def handle_add_tracking(call):
+        """Call when a user adds a new Aftership tracking from HASS."""
+        title = call.data.get(CONF_TITLE)
+        slug = call.data.get(CONF_SLUG)
+        tracking_number = call.data[CONF_TRACKING_NUMBER]
+
+        await aftership.add_package_tracking(tracking_number, title, slug)
+        await update(instance)
```

Comment:
```suggestion
        await instance.async_update()
```

Reply:
Awesome! I hated my solution to get around the throttle. Thank you!